gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFARecoveryCodeEndpoint.java

```diff
@@ -178,4 +195,43 @@ private boolean failIfUserIsNotPresent(RoutingContext routingContext){
 
         return false;
     }
+
+    private Single<EnrolledFactorSecurity> generateRecoveryCode(io.gravitee.am.model.User endUser, Client client) {
+        final Factor recoveryFactor = getClientRecoveryFactor(client);
+        final FactorProvider recoveryFactorProvider = factorManager.get(recoveryFactor.getId());
+        final Map<String, Object> factorData = Map.of(
+                FactorContext.KEY_RECOVERY_FACTOR,
+                recoveryFactor, KEY_USER, endUser);
+        final FactorContext recoveryFactorCtx = new FactorContext(applicationContext, factorData);
+
+        return ((RecoveryFactor) recoveryFactorProvider).generateRecoveryCode(recoveryFactorCtx);
```

Comment:
NPE with `getClientRecoveryFactor(client)`

Reply:
fixed the NPE issue